runtime/lua/vim/ui.lua

```diff
@@ -87,7 +87,9 @@ function M.input(opts, on_confirm)
     on_confirm = { on_confirm, 'function', false },
   })
 
-  opts = opts or {}
+  if not opts or vim.tbl_isempty(opts) then
+    opts = vim.empty_dict()
+  end
```

Comment:
Ternaries can help readability because they avoid nesting and helps the reader see that an assignment is being done to the current scope. 
```suggestion
  opts = (opts and not vim.tbl_isempty(opts)) and opts or vim.empty_dict()
```

Wonder why tbl_isempty doesn't accept nil ...

Reply:
oh nice! I was trying to do this but I didn't have the `and opts`